Renaming DoStructureMigration -> DataObjectMigration

stephan-merkli opened the pull request:
Because the `DoStructureMigrator` executes value migrations too (`IDoValueMigrationHandler`), several renamings were applied.

315763

LukasHuser commented on org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/migration/fixture/house/PetFixtureFamilyFriendlyMigrationHandlerInvalidTypeVersionToUpdate_3.java:
```diff
@@ -15,9 +15,9 @@
 import org.eclipse.scout.rt.dataobject.ITypeVersion;
 import org.eclipse.scout.rt.dataobject.migration.AbstractDoStructureMigrationHandler;
-import org.eclipse.scout.rt.dataobject.migration.DoStructureMigrationContext;
+import org.eclipse.scout.rt.dataobject.migration.DataObjectMigrationContext;
```
Organize imports over all affected classes required? Import order might change.

stephan-merkli replied:
Done

stephan-merkli committed:
Renaming DoStructureMigration -> DataObjectMigration

Because the `DoStructureMigrator` executes value migrations too (`IDoValueMigrationHandler`), several renamings were applied.

315763
